e2e-tests/src/npg/psp-sorting.npg.integration.test.ts

```diff
@@ -0,0 +1,106 @@
+import { choosePaymentMethod, fillEmailForm, fillOnlyCardDataForm, fillPaymentNotificationForm } from "./helpers";
+import { selectLanguage } from "../verify/helpers";
+
+/**
+ * Test input and configuration
+ */
+const CHECKOUT_URL = String(process.env.CHECKOUT_URL);
+const VALID_FISCAL_CODE = String(process.env.VALID_FISCAL_CODE);
+const EMAIL = String(process.env.EMAIL);
+
+/**
+ * Increase default test timeout (120000ms)
+ * to support entire payment flow
+  */
+
+const timeout = 80_000
+jest.setTimeout(timeout);
+jest.retryTimes(2);
+page.setDefaultNavigationTimeout(timeout);
+page.setDefaultTimeout(timeout)
+
+beforeAll( async () => {
+  await page.goto(CHECKOUT_URL);
+  await page.setViewport({ width: 1200, height: 907 });
+})
+
+beforeEach(async () => {
+  await page.goto(CHECKOUT_URL);
+  await selectLanguage("it");
+});
+
+
+describe("Checkout show and sort psp list", () => {
+
+  it("Should show psp sorted list", async() => {
+   
+    // notice code with 6000 euro as amount (according Mock EC)
+    const CODICE_AVVISO = "30202" + Math.floor(Math.pow(10, 12) + Math.random() * 9 * Math.pow(10, 12))
+
+    console.log(
+      `Testing psp list is sorted .
+      notice code: ${CODICE_AVVISO},
+      fiscal code: ${VALID_FISCAL_CODE}
+      `,
+    );
+
+    const payNoticeBtnSelector = '#paymentSummaryButtonPay';
+    const pspEditButtonSelector = "#pspEdit";
+    const pspFeeSortButtonId = "#sortByFee";
+
+    await fillPaymentNotificationForm(CODICE_AVVISO, VALID_FISCAL_CODE);
+    const payNoticeBtn = await page.waitForSelector(payNoticeBtnSelector);
+    await payNoticeBtn.click();
+    await fillEmailForm(EMAIL);
+    await choosePaymentMethod('CP');
+    await fillOnlyCardDataForm({
+      number: "4242424242424242",
+      expirationDate: "1230",
+      ccv: "123",
+      holderName: "Test test"
+    })
+    const pspEditButton = await page.waitForSelector(pspEditButtonSelector);
+    await pspEditButton.click();
+    await new Promise(r => setTimeout(r, 1000));
```

Comment:
You could remove this timeout by disable animations with this code on beforeAll function:

```
page.on("load", () => {
    const content = `* {transition-duration: 0s !important;}`;
    page.addStyleTag({ content });
});
```
With this change pspList should be visible immediately

Reply:
here there 2 kind of delay to be addressed:
1. animation delay. Your solution could be useful, even if we are mocking a behavior we would to be more similar to user experience
2. network delay: the psp list is shown immediately, and psp list could be loaded some milliseconds after the list is shown. In some cases not all data are really available when calculate fees responses (maybe is a minor rendering issue). Anyway I tried it and it seems to not work fine every time, so, if you don't mind, i would mantain the timeout. 

We could consider to address the timeout cluttering in a dedicated pr, to improve our tests reliabilty

Reply:
If removing the animations does not solve all the problems for me it's okay how you did, also because it is an acceptable timeout